web/res/js/donationForm.js

```diff
@@ -25,7 +25,7 @@ $( function () {
 			} ),
 			WMDE.Components.createRadioComponent( store, $( '.address-type-select' ), 'addressType' ),
 			WMDE.Components.createRadioComponent( store, $( '.salutation' ), 'salutation' ),
-			WMDE.Components.createRadioComponent( store, $( '.personal-title' ), 'title' ),
+			WMDE.Components.createSelectMenuComponent( store, $( '.personal-title' ), 'title' ),
```

Comment:
The issue was that we use the "selectmenu" jQuery UI component for all `select` fields. While the `RadioComponent` works for normal select fields, for the jquery ui selectmenu it doesn't, because the `change` event is not triggered and this the store (that stores all client side state) is not updated. This lead to a reset of the title field whenever a different field was changed (because all field values are updated with the values from the store whenever a value changes). The `SelectMenuComponent` listens to the `selectmenuchange` event and updates the store accordingly.